File: lang/script/ruby/fizzbuzz3.rb
#! /usr/bin/env ruby
# coding: utf-8

a = Array.new( 101 )
for i in (1..100)
  a[i] = i
end

s = { 3 => "Fizz", 5 => "Buzz", 15 => "FizzBuzz" }
for n in [3, 5, 15]
  i = n
  f = s[n]
  while i <= 100
    a[i] = f
    i += n
  end
end
p a.pop( 100 )		# 先頭の要素（a[0]）を除く

File: lang/script/go/fizzbuzz3.go
package main
import "fmt"
var (
	a = [101]string{}
	k = [3]int{ 3, 5, 15 }
	s = [3]string{ "Fizz", "Buzz", "FizzBuzz" }
)
func main() {
	for i := 1; i <= 100; i++ {
		a[i] = fmt.Sprint( i )
	}
	for i := 0; i < 3; i++ {
		n := k[i]
		t := s[i]
		for j := n; j <= 100; j += n {
			a[j] = t
		}
	}
	for i := 1; i <= 100; i++ {
		fmt.Print( a[i], " " )
	}
	fmt.Println( "" )
}
